### ISSUE TITLE
Retrain bot

### ISSUE THREAD

**Issue by saveSession** [Conchylicultor/DeepQA#62]
There is a way to retrain the bot with another dataset or update the actual dataset? I tried to change the dataset file, but it seems to be incompatible because the word embedding, getting this error "Assign requires shapes of both tensors to match. lhs shape=[38746]  rhs shape= [58311]. ". 

Thanks!
(I'm just learning, sorry for the inconvenience)

**Comment by Conchylicultor** [comment]
You are right, the models are incompatibles between the datasets. It is a limitation of the program. A more serious problem is that the models are even incompatible within the same dataset if the sentenceLength parameter is different.

This is due to the fact that each dataset has a different vocabulary. To retrain a model on another dataset, you should change the vocabulary of the new dataset to match the one from the old one and replace all extra words by the `UNKNOWN` token. Basically, that require to re-precompute the new dataset, but by using the vocabulary from the old one.

If you really need to do it, here are the steps.

Step 1: force the loading corpus of the old dataset on which the model has been trained. Those two variables contains the vocabulary. https://github.com/Conchylicultor/DeepQA/blob/master/chatbot/textdata.py#L278

        self.word2id = data['word2id']
        self.id2word = data['id2word']

Step 2: Regenerate the new dataset with the old vocabulary. The only modification in the program should be at this line: https://github.com/Conchylicultor/DeepQA/blob/master/chatbot/textdata.py#L347, by adding the param `create=False`, you can desactivate the vocabulary creation.

       tempWords.append(self.getWordId(token, create=False))  # Create the vocabulary and the training sentences

You can check if the dataset has been correctly create by playing some samples with the `--playDataset` flag. You should see the words from the first dataset with some `UNKNOWN` words for the words of the new dataset which don't belong to the first one.

**Comment by shamanez** [comment]
I cannot over-fit the model for the lighweight data set. I think it's more to do with the dialog text file.
